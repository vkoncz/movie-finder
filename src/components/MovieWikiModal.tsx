import React from 'react';

interface Props {
  movieTitle: string;
}

export const MovieWikiModal: React.FC<Props> = ({ movieTitle }) => {
  return <div>Movie modal for {movieTitle}</div>;
};
