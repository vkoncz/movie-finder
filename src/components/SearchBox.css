ion-button {
  height: 42px;
  width: 60px;
}

ion-grid {
  max-width: 1000px;
}
