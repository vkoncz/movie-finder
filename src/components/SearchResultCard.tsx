import {
  IonCard,
  IonCardContent,
  IonCardHeader,
  IonCardTitle,
  IonIcon,
  IonItem,
  IonLabel,
} from '@ionic/react';
import React from 'react';
import { Details } from '../models/MovieDetails';
import { star, pricetag } from 'ionicons/icons';

interface Props {
  details: Details;
  onMovieClick: (title: string) => void;
}

export const SearchResultCard: React.FC<Props> = ({ details, onMovieClick }) => {
  return (
    <IonCard
      button={true}
      onClick={() => {
        onMovieClick(details.title);
      }}
    >
      <IonCardHeader color="medium">
        <IonCardTitle>{details.title}</IonCardTitle>
      </IonCardHeader>

      <IonCardContent>
        <IonItem lines="none">
          <IonIcon icon={star} slot="start" size="small"></IonIcon>
          <IonLabel>{details.rating}</IonLabel>
        </IonItem>
        <IonItem lines="none">
          <IonIcon icon={pricetag} slot="start" size="small"></IonIcon>
          <IonLabel>{details.genres.map(genre => genre.name).join(', ')}</IonLabel>
        </IonItem>
      </IonCardContent>
    </IonCard>
  );
};
