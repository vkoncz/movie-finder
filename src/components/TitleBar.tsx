import React from 'react';
import { IonTitle, IonToolbar } from '@ionic/react';

export const TitleBar: React.FC = () => {
  return (
    <IonToolbar color="primary">
      <IonTitle>Movie Finder</IonTitle>
    </IonToolbar>
  );
};
