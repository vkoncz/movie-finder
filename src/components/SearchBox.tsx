import React, { useState } from 'react';
import {
  IonButton,
  IonCol,
  IonRow,
  IonGrid,
  IonIcon,
  IonSearchbar,
  IonToast,
  IonLoading,
} from '@ionic/react';
import { search, videocam } from 'ionicons/icons';
import './SearchBox.css';
import { Details } from '../models/MovieDetails';
import { graphSearchClient } from '../data/fetch/graphSearchClient';

interface Props {
  resultChange: (results?: Details[]) => void;
}

export const SearchBox: React.FC<Props> = ({ resultChange }) => {
  const [searchTerm, setSearchTerm] = useState('');
  const [showError, setShowError] = useState(false);
  const [isLoading, setIsLoading] = useState(false);

  const fetchData = () => {
    setIsLoading(true);
    graphSearchClient(searchTerm)
      .then(result => {
        resultChange(result);
        setIsLoading(false);
      })
      .catch(error => {
        console.log('Connection error:', error);
        setShowError(true);
        setIsLoading(false);
      });
  };

  return (
    <>
      <IonGrid class="ion-center search-container">
        <IonRow class="ion-align-items-center">
          <IonCol class="ion-no-padding">
            <IonSearchbar
              autocomplete="on"
              placeholder="Search movies"
              searchIcon={videocam}
              value={searchTerm}
              onIonChange={e => setSearchTerm(e.detail.value!)}
              onKeyUp={e => {
                if (e.key === 'Enter') fetchData();
              }}
            />
          </IonCol>
          <IonCol size="auto" class="ion-no-padding ion-align-self-center">
            <IonButton onClick={fetchData} class="search-button">
              <IonIcon icon={search} />
            </IonButton>
          </IonCol>
        </IonRow>
      </IonGrid>

      <IonLoading isOpen={isLoading} message="Searching movies..." />

      <IonToast
        isOpen={showError}
        onDidDismiss={() => setShowError(false)}
        message="Connection error, please try it again later"
        duration={5000}
      />
    </>
  );
};
