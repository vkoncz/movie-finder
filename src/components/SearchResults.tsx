import { IonCol, IonGrid, IonModal, IonRow, IonText } from '@ionic/react';
import React, { useState } from 'react';
import { Details } from '../models/MovieDetails';
import { MovieInfo } from '../models/MovieInfo';
import { MovieWikiModal } from './MovieWikiModal';
import { SearchResultCard } from './SearchResultCard';

interface Props {
  results?: Details[];
}

export const SearchResults: React.FC<Props> = ({ results }) => {
  const message = !results || results?.length ? '' : 'No results found';
  const [selectedMovieInfo, setSelectedMovieInfo] = useState<MovieInfo>({ imdb: '', title: '' });
  const [modalVisible, setModalVisible] = useState(false);

  return (
    <>
      <IonGrid>
        <IonRow>
          {results
            ? results.map((result, key) => (
                <IonCol key={key} sizeXl="3" sizeLg="4" sizeMd="6" size="12">
                  <SearchResultCard
                    details={result}
                    onMovieClick={title => {
                      setModalVisible(true);
                      setSelectedMovieInfo(title);
                    }}
                  />
                </IonCol>
              ))
            : ''}
          <IonText>{message}</IonText>
        </IonRow>
      </IonGrid>

      <IonModal isOpen={modalVisible} onDidDismiss={() => setModalVisible(false)}>
        <MovieWikiModal movieInfo={selectedMovieInfo} modalClose={() => setModalVisible(false)} />
      </IonModal>
    </>
  );
};
