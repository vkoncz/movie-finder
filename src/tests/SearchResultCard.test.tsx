import React from 'react';
import { render } from '@testing-library/react';
import { SearchResultCard } from '../components/SearchResultCard';
import { details } from './test-data/details';

test('should match to SearchResultCard snapshots', () => {
  const { asFragment } = render(<SearchResultCard details={details} onMovieClick={() => {}} />);

  expect(asFragment()).toMatchSnapshot();
});
