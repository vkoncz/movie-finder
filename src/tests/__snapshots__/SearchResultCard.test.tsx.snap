// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match to SearchResultCard snapshots 1`] = `
<DocumentFragment>
  <ion-card>
    <ion-card-header
      color="medium"
    >
      <ion-card-title>
        The Lord of the Rings: The Return of the King
      </ion-card-title>
    </ion-card-header>
    <ion-card-content>
      <ion-item
        lines="none"
      >
        <ion-icon
          icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Star</title><path d='M394 480a16 16 0 01-9.39-3L256 383.76 127.39 477a16 16 0 01-24.55-18.08L153 310.35 23 221.2a16 16 0 019-29.2h160.38l48.4-148.95a16 16 0 0130.44 0l48.4 149H480a16 16 0 019.05 29.2L359 310.35l50.13 148.53A16 16 0 01394 480z'/></svg>"
          size="small"
          slot="start"
        />
        <ion-label>
          8.5
        </ion-label>
      </ion-item>
      <ion-item
        lines="none"
      >
        <ion-icon
          icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Pricetag</title><path d='M467 45.2A44.45 44.45 0 00435.29 32H312.36a30.63 30.63 0 00-21.52 8.89L45.09 286.59a44.82 44.82 0 000 63.32l117 117a44.83 44.83 0 0063.34 0l245.65-245.6A30.6 30.6 0 00480 199.8v-123a44.24 44.24 0 00-13-31.6zM384 160a32 32 0 1132-32 32 32 0 01-32 32z'/></svg>"
          size="small"
          slot="start"
        />
        <ion-label>
          Adventure, Fantasy, Action
        </ion-label>
      </ion-item>
    </ion-card-content>
  </ion-card>
</DocumentFragment>
`;
